decoder/iface_name_test.go

```diff
@@ -0,0 +1,45 @@
+package decoder
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/cloudflare/ebpf_exporter/v2/config"
+)
+
+func TestIfNameDecoder(t *testing.T) {
+	cases := []struct {
+		in    []byte
+		cache map[uint32][]byte
+		out   []byte
+	}{
+		{
+			in:    []byte{0x1, 0x0, 0x0, 0x0},
+			cache: map[uint32][]byte{uint32(1): []byte("lo"), uint32(2734686214): []byte("eth42")},
```

Comment:
I think you can introduce a `cache` variable for this and reference it here and below.